src/net.cpp

```diff
@@ -1166,16 +1166,27 @@ void CConnman::CreateNodeFromAcceptedSocket(SOCKET hSocket,
 
 bool CConnman::AddConnection(const std::string& address, ConnectionType conn_type)
 {
-    if (conn_type != ConnectionType::OUTBOUND_FULL_RELAY && conn_type != ConnectionType::BLOCK_RELAY) return false;
-
-    const int max_connections = conn_type == ConnectionType::OUTBOUND_FULL_RELAY ? m_max_outbound_full_relay : m_max_outbound_block_relay;
+    int max_connections;
+    switch(conn_type) {
+        case ConnectionType::INBOUND:
+        case ConnectionType::MANUAL:
+        case ConnectionType::FEELER:
+            return false;
+        case ConnectionType::OUTBOUND_FULL_RELAY:
+            max_connections = m_max_outbound_full_relay;
+        case ConnectionType::BLOCK_RELAY:
+            max_connections = m_max_outbound_block_relay;
+        // no limit for ADDR_FETCH because -seednode has no limit either
+        case ConnectionType:: ADDR_FETCH:
+            max_connections = 0;
+    }
```

Comment:
perhaps give `max_connections` a value on initialization and add a comment per doc/developer-notes.md
```diff
 bool CConnman::AddConnection(const std::string& address, ConnectionType conn_type)
 {
-    int max_connections;
+    int max_connections{0};
     switch(conn_type) {
         case ConnectionType::INBOUND:
         case ConnectionType::MANUAL:
@@ -1178,8 +1178,8 @@ bool CConnman::AddConnection(const std::string& address, ConnectionType conn_typ
             max_connections = m_max_outbound_block_relay;
         // no limit for ADDR_FETCH because -seednode has no limit either
         case ConnectionType:: ADDR_FETCH:
-            max_connections = 0;
-    }
+            break;
+    } // no default case, so the compiler can warn about missing cases
```

Reply:
done